### PR TITLE
Disable showing vertical banners in profile picker dialog

### PR DESCRIPTION
**Pull request by emerick**
Resolves https://github.com/brave/brave-browser/issues/30599

## Submitter Checklist:

- [ ] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: patches/chrome-browser-resources-signin-profile_picker-profile_picker_app.ts.patch [hunk 1]
```diff
@@ -0,0 +1,12 @@
+diff --git a/chrome/browser/resources/signin/profile_picker/profile_picker_app.ts b/chrome/browser/resources/signin/profile_picker/profile_picker_app.ts
+index 4138be7976b442c797e76d152e683d4d9da25635..c4179b55214a65b38cb5d1bc3dc40c0d21b4f9e9 100644
+--- a/chrome/browser/resources/signin/profile_picker/profile_picker_app.ts
++++ b/chrome/browser/resources/signin/profile_picker/profile_picker_app.ts
+@@ -179,6 +179,7 @@ export class ProfilePickerAppElement extends ProfilePickerAppElementBase {
+     } else {
+       this.shouldDisplayVerticalBanners_ = false;
+     }
++    this.shouldDisplayVerticalBanners_ = false;
```
Comment (bridiver) [inline patches/chrome-browser-resources-signin-profile_picker-profile_picker_app.ts.patch:9]: isn't whatever image they are using there part of branding so we can replace it?

Comment (bridiver) [reply patches/chrome-browser-resources-signin-profile_picker-profile_picker_app.ts.patch:9]: or if not, can we override the style using an existing patch?

Comment (bridiver) [reply patches/chrome-browser-resources-signin-profile_picker-profile_picker_app.ts.patch:9]: looks like it's not part of branding which is really odd, but a more extensible patch would be to replace it `static_files` `chrome/browser/resources/signin`

**Review by mkarolin — APPROVED** [review]
LGTM

**Review by bridiver — APPROVED** [review]

**Comment by stephendonner** [comment]
## Verified `PASSED` using 

Brave | 1.54.11 Chromium: 114.0.5735.53 (Official Build) nightly (x86_64)
-- | --
Revision | c499d7ea22c8b2dba278465a5df7b86a8efa4e64-refs/branch-heads/5735@{#970}
OS | macOS Version 11.7.7 (Build 20G1345)

### Steps:
1. installed `1.54.11`
2. launched Brave
3. clicked on the "hamburger" menu
4. clicked on `Create a New Profile`
5. compared to `1.51.114`

### Confirmed the `Chrome`-specific background was removed

`1.54.11` | `1.51.114`
---------|----------
Screen Shot 2023-05-28 at 12 20 03 PM | Screen Shot 2023-05-28 at 3 10 26 AM

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Win 11 x64` using the following builds:

```
Brave | 1.54.15 Chromium: 114.0.5735.53 (Official Build) nightly (64-bit)
-- | --
Revision | c499d7ea22c8b2dba278465a5df7b86a8efa4e64-refs/branch-heads/5735@{#970}
OS | Windows 11 Version 22H2 (Build 22621.1702)
```

* ensured that the Chrome vertical banner is removed a clean profile/install
* ensured that the Chrome vertical banner is removed from an upgraded profile
* ensured that `Guest Mode` from the profile picker still works without issues
* ensured that creating new profiles within the profile picker works without issues
* ensured that switching between different profiles within the profile picker works without issues

**`Upgrade`** | **`Clean Install/Profile`**
----------------|-----------------------------
UpgradeProfile | newProfile

**Pull request merged**